kernel/aster-nix/src/fs/file_table.rs

```diff
@@ -240,13 +279,30 @@ impl FileTableEntry {
             file,
             flags: AtomicU8::new(flags.bits()),
             subject: Subject::new(),
+            owner: RwLock::new(None),
         }
     }
 
     pub fn file(&self) -> &Arc<dyn FileLike> {
         &self.file
     }
 
+    pub fn owner(&self) -> Option<Pid> {
+        self.owner.read().as_ref().map(|(pid, _)| *pid)
+    }
+
+    pub fn set_owner(&self, owner: Pid) -> Result<()> {
+        if let Some((_, observer)) = self.owner.read().as_ref() {
+            let _ = self.file.unregister_observer(&Arc::downgrade(observer));
+        }
+
+        let observer = OwnerObserver::new(self.file.clone(), owner);
+        self.file
+            .register_observer(observer.weak_self(), IoEvents::empty())?;
+        *self.owner.write() = Some((owner, observer));
```

Comment:
This method is not implemented in a thread-safe way as the lock-protected scope is too small. Think about what would happen if `set_owner` is called concurrently.